src/gui/TimeLineWidget.cpp

```diff
@@ -224,52 +227,59 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 {
 	QPainter p( this );
 
-	QColor bg_color = QApplication::palette().color( QPalette::Active,
-							QPalette::Background );
-	QLinearGradient g( 0, 0, 0, height() );
-	g.setColorAt( 0, bg_color.lighter( 150 ) );
-	g.setColorAt( 1, bg_color.darker( 150 ) );
-	p.fillRect( 0, 0, width(), height(), g );
+	// Draw background
+	p.fillRect( 0, 0, width(), height(), p.background() );
 
+	// Clip so that we only draw everything starting from the offset
 	p.setClipRect( m_xOffset, 0, width() - m_xOffset, height() );
-	p.setPen( QColor( 0, 0, 0 ) );
 
-	p.setOpacity( loopPointsEnabled() ? 0.9 : 0.2 );
-	p.drawPixmap( markerX( loopBegin() )+2, 2, *s_loopPointBeginPixmap );
-	p.drawPixmap( markerX( loopEnd() )+2, 2, *s_loopPointEndPixmap );
-	p.setOpacity( 1.0 );
+	// Draw the bar lines and numbers
+	// Activate hinting on the font
+	QFont font = p.font();
+	font.setHintingPreference( QFont::PreferFullHinting );
+	p.setFont(font);
 
+	QColor const & barLineColor = getBarLineColor();
+	QColor const & barNumberColor = getBarNumberColor();
 
-	tact_t tact_num = m_begin.getTact();
-	int x = m_xOffset + s_posMarkerPixmap->width() / 2 -
+	tact_t barNumber = m_begin.getTact();
+	int const x = m_xOffset + s_posMarkerPixmap->width() / 2 -
 			( ( static_cast<int>( m_begin * m_ppt ) / MidiTime::ticksPerTact() ) % static_cast<int>( m_ppt ) );
 
-	QColor lineColor( 192, 192, 192 );
-	QColor tactColor( lineColor.darker( 120 ) );
-
-	// Set font to half of the widgets size (in pixels)
-	QFont font = p.font();
-	font.setPixelSize( this->height() * 0.5 );
-	p.setFont( font );
-
 	for( int i = 0; x + i * m_ppt < width(); ++i )
 	{
-		const int cx = x + qRound( i * m_ppt );
-		p.setPen( lineColor );
-		p.drawLine( cx, 5, cx, height() - 6 );
-		++tact_num;
-		if( ( tact_num - 1 ) %
+		++barNumber;
+		if( ( barNumber - 1 ) %
 			qMax( 1, qRound( 1.0f / 3.0f *
 				MidiTime::ticksPerTact() / m_ppt ) ) == 0 )
 		{
-			const QString s = QString::number( tact_num );
-			p.setPen( tactColor );
-			p.drawText( cx + qRound( ( m_ppt - p.fontMetrics().
-							width( s ) ) / 2 ),
-				height() - p.fontMetrics().ascent() / 2, s );
+			const int cx = x + qRound( i * m_ppt );
+			p.setPen( barLineColor );
+			p.drawLine( cx, 5, cx, height() - 6 );
+
+			const QString s = QString::number( barNumber );
+			p.setPen( barNumberColor );
+			p.drawText( cx + 5, height() - p.fontMetrics().ascent() / 2, s );
 		}
 	}
 
+	// Draw the loop rectangle
+	int const loopRectMargin = 1;
+	int const loopRectHeight = this->height() - 2 * loopRectMargin;
+	int const loopStart = markerX( loopBegin() ) + 8;
+	int const loopEndR = markerX( loopEnd() ) + 9;
+	int const loopRectWidth = loopEndR - loopStart;
+	p.setPen(Qt::NoPen);
+	p.setBrush(loopPointsEnabled() ? getActiveLoopColor() : getInactiveLoopColor());
```

Comment:
You don't have space between parentheses on these 3 lines, while the rest of the file looks differently. It'd be nice if you changed it for consistency.